### PR TITLE
Mike Kerbleski

### PR DESCRIPTION
**Pull request by MKerbleski**

### CODE REVIEW

File: config/routes.js [hunk 1]
```diff
@@ -1,19 +1,74 @@
...
+    jwtid:'54321',
+  }
+  return jwt.sign(payload, jwtKey, options)
+}
+
+
 function register(req, res) {
   // implement user registration
+  const user = req.body
+  const hash = bcrypt.hashSync(user.password, 4);
```
Comment (adfaris) [inline config/routes.js]: why did you pick a salt value of 4?

Comment (MKerbleski) [reply config/routes.js]: I figured 10+ would be excessive for this exercise

File: ANSWERS.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
 <!-- Answers to the Short Answer Essay Questions go here -->
 
 1. What is the purpose of using _sessions_?
+    the purpose of sessions is to gather and store user informations
```
Comment (adfaris) [inline ANSWERS.md]: the purpose of sessions is not only to store information. Here is a reading material http://www.lassosoft.com/Tutorial-Understanding-Cookies-and-Sessions

File: ANSWERS.md [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
 2. What does bcrypt do to help us store passwords in a secure manner.
+    bcrypt helps store password by storing a hash of the password and never the password itself. Using 1 way cryptographic hashs allows us to validate that the correct password was input with only a refrences. 
 
 3. What does bcrypt do to slow down attackers?
+    bcrypt uses 'salt' a string that is added to the users password to further secure the password in case a hacker got ahold of the hashed passwords.
```
Comment (adfaris) [inline ANSWERS.md]: Salt is the number that you pass to bcrypt to has it the salt amount times, it is calacuated by the power of 2, if you pass salt value of 4 for salt, you are hashing 2^4 = 16times.

File: config/routes.js [hunk 4]
```diff
@@ -1,19 +1,75 @@
...
+        const token = generateToken(dbUser)
+        res.status(200).json({message: 'user logged in. token created', token: token})
+      } else {
+        res.status(400).json({message: 'invalid username or password'})
+      }
+    }).catch(err => {
+      res.status(400).send({message: 'username or password do not match records'})
+    }
+    )
+
```
Comment (adfaris) [inline config/routes.js]: Good job implementing the login and register.

File: da_jokes/src/components/register.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                        <button>Register</button>
+                    </form>
+                </RegisterDiv>
+            </React.Fragment>
+        )
+    }
+}
+
+export default Register;
+
```
Comment (adfaris) [inline da_jokes/src/components/register.js]: Your register page looks good.

**Pull request closed without merging**